dev/integration_tests/wide_gamut_test/lib/main.dart

```diff
@@ -77,27 +87,100 @@ class MyApp extends StatelessWidget {
       theme: ThemeData(
         primarySwatch: Colors.blue,
       ),
-      home: const MyHomePage(title: 'Wide Gamut Test'),
+      home: MyHomePage(_setup, title: 'Wide Gamut Test'),
     );
   }
 }
 
-class MyHomePage extends StatelessWidget {
-  const MyHomePage({super.key, required this.title});
+class _SaveLayerDrawer extends CustomPainter {
+  final ui.Image? _image;
+  _SaveLayerDrawer(this._image);
+
+  @override
+  void paint(Canvas canvas, Size size) {
+    if (_image != null) {
+      final Rect imageRect = Rect.fromCenter(
+          center: const Offset(0, 0),
+          width: _image!.width.toDouble(),
+          height: _image!.height.toDouble());
+      canvas.saveLayer(
+          imageRect,
+          Paint());
+      canvas.drawRect(
+          imageRect.inflate(-_image!.width.toDouble() / 4.0),
+          Paint()
+            ..style = PaintingStyle.stroke
+            ..color = const Color(0xff00ff00)
+            ..strokeWidth = 3);
+      canvas.saveLayer(
+          imageRect,
+          Paint()..blendMode = BlendMode.dstOver);
+      canvas.drawImage(_image!,
+          Offset(-_image!.width / 2.0, -_image!.height / 2.0), Paint());
+      canvas.restore();
+      canvas.restore();
```

Comment:
Impeller doesn't do so today, but I think it would be possible to optimize the second savelayer away by just reversing the draw order.

A reasonably future proof way to force a savelayer would be to draw 4 overlapping objects:
* Draw two blended things in the background,
* append a savelayer that applies using a second blend mode,
* and draw two blended things within the savelayer that use a third blend mode.

For example:
1. Draw an image in the background
2. Cover the whole image with a partially translucent rectangle (with just kSrcOver, or any non-destructive blend mode)
3. Push a savelayer with a non-separable blend mode such as kHue or kLuminosity
4. Draw two partially overlapping circles within the image area using a simple separable blend mode, such as kPlus

Impeller basically has no choice but to render _something_ offscreen for this, regardless of what nutty optimizations we manage to come up with.

Reply:
Are you sure the offscreen buffer could be optimized away?  `dstOver` is a raster operation.  It can't be calculated without having raster images which require offscreen buffers to create.  I know we can look at the results and say there is an alternative way to get the same result.  I'm not sure it's feasible to create such an optimization at runtime though.

Reply:
> Are you sure the offscreen buffer could be optimized away?

The first savelayer is almost a no-op because the background is empty and the blend mode is `srcOver`. It does have a bounds hint, but that can be handled by inserting a clip rect.

The second savelayer only contains one `srcOver` draw within it. In this situation we can always remove the savelayer and replace the one draw call's blend mode with the savelayer's blend mode.

> `dstOver` is a raster operation. It can't be calculated without having raster images which require offscreen buffers to create.

`dstOver` (and all of the other "Porter-Duff" composites) are implemented using the [pipeline blend config](https://github.com/flutter/engine/blob/c4f51bc78644a3e120862e614cbe13e50b6187b8/impeller/entity/contents/content_context.cc#L60-L65) in Impeller, and so the blend source doesn't actually need to be rendered to a separate texture first (see [this doc](https://github.com/flutter/engine/blob/main/impeller/docs/blending.md) for more deets about the blending situation).

Reply:
Thanks, reading through the docs it looks like `multiply` is not able to be optimized similarly since it is operating on the colors, not just the alpha values.  So I've switched the test to use `multiply`.  I think that solves the problem, let me know if I'm missing something.

Reply:
The second savelayer can be optimized away regardless of the blend mode.
And for some targets (Vulkan and OpenGL with common extensions), well be adding a path to implement all of the blend modes as pipeline ops.

If you don't care about testing the savelayer path specifically and just want to force a scenario where something draws to an offscreen texture, a safer option would be to blur an image with a reasonably large sigma.